sqlmesh/core/snapshot/evaluator.py

```diff
@@ -407,15 +411,16 @@ def close(self) -> None:
     def _create_snapshot(
         self,
         snapshot: Snapshot,
-        snapshots: t.Dict[SnapshotId, Snapshot],
+        snapshots: t.Collection[Snapshot],
         deployability_index: t.Optional[DeployabilityIndex],
         on_complete: t.Optional[t.Callable[[SnapshotInfoLike], None]],
     ) -> None:
         if not snapshot.is_model:
             return
 
+        snapshots_by_id = {s.snapshot_id: s for s in snapshots}
```

Comment:
It seems like after we switched to Collection everywhere we keep recreating this dict over and over again. Please correct me if I'm wrong.

Reply:
I like the previous approach better, because `dict.values()` doesn't really construct a new collection but returns a view instead.